news/_posts/2020-10-09-how-to-write-a-new-ui-test.md

```diff
@@ -0,0 +1,407 @@
+---
+layout: post
+title:  "Automated UI tests tutorial"
+subtitle: "Step by step - Adding a new PrestaShop UI test"
+date:  2020-10-09 12:00:00
+authors: [boubkerbribri]
+icon: icon-github-alt
+image: /assets/images/theme/meta-logo-build.png
+tags: [community, contribution, tests, automation]
+---
+
+One year ago, the QA team started to develop a new test framework, based on [Puppeteer](https://github.com/puppeteer/puppeteer), [Mocha](https://mochajs.org/), and [Chai](https://www.chaijs.com/). You can read more about the framework in this previous article [The (new) PrestaShop Test Framework](https://build.prestashop.com/news/the-new-prestashop-test-framework/).
+Since then, the tests coverage has been continually increasing, and we recently improved our framework by switching to [Playwright](https://playwright.dev/).
+
+In this article, we will explain how to create a new UI (User Interface) test for the [PrestaShop core project](https://github.com/PrestaShop/PrestaShop).
+
+Before we start, you should definitely read about the stack, and the architecture chosen for this project in the [PrestaShop devdoc](https://devdocs.prestashop.com/1.7/testing/ui-tests/how-to-contribute-and-create-ui-tests/). 
+
+
+## I. Writing the scenario
+
+The first step to writing your scenario is to identify exactly what you want to test, and limit the test's scope to that. For this example **we will be checking that the customer link on "Orders" page (on the Back Office) redirects to the "view customer" page**.
+
+The second step involves a manual check of this very scenario to be sure it's working, and to write down all the steps needed, from start to finish:
+
+- Log in to the Back Office
+- Go to the 'Orders' page
+- Reset all filters
+- Filter orders by customer name
+- Click on customer link on grid
+- Check 'View customer' page is displayed
+ 
+Now, we have the scenario, we can create a new javascript file and write our scenario using Mocha (see example below).
+The directory in which we create our file should be chosen wisely (which campaign? BO or FO? Which page in BO? ...). This file organization is very important since Mocha lets you run test by folders (recursively or not), so grouping your tests by features or meta-features (pages) is a very good idea.
```

Comment:
```suggestion
The directory in which we create our file should be chosen wisely (Which campaign? BO or FO? Which page in BO? ...). This file organization is very important since Mocha lets you run test by folders (recursively or not), so grouping your tests by features or meta-features (pages) is a very good idea.
```

Reply:
done